miden/src/cli/run.rs

```diff
@@ -100,6 +104,18 @@ impl RunCmd {
 // HELPER FUNCTIONS
 // ================================================================================================
 
+pub fn create_execution_options(cmd: &RunCmd) -> Result<ExecutionOptions, Report> {
+    let mut execution_options =
+        ExecutionOptions::new(Some(cmd.max_cycles), cmd.expected_cycles, cmd.tracing)
+            .into_diagnostic()?;
+
+    if cmd.debug {
+        execution_options = execution_options.with_debugging();
+    }
+
+    Ok(execution_options)
+}
+
 #[instrument(name = "run_program", skip_all)]
 fn run_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
     // load libraries from files
```

Comment:
On line 126 below, we should pass `Debug::On` as an argument to `compile()` when `--debug` flag is specified. This should enable program compilation in debug mode (and you could avoid modifying the assembler).